conan_provider.cmake

```diff
@@ -19,18 +17,53 @@ function(detect_os OS OS_API_LEVEL OS_VERSION)
             message(STATUS "CMake-Conan: android_platform=${ANDROID_PLATFORM}")
             set(${OS_API_LEVEL} ${_OS_API_LEVEL} PARENT_SCOPE)
         endif()
+        if(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin" OR
+           ${CMAKE_SYSTEM_NAME} STREQUAL "iOS" OR
+           ${CMAKE_SYSTEM_NAME} STREQUAL "tvOS" OR
+           ${CMAKE_SYSTEM_NAME} STREQUAL "watchOS")
+            if(DEFINED CMAKE_OSX_SYSROOT)
+                # CMAKE_OSX_SYSROOT contains the full path to the SDK for MakeFile/Ninja
```

Comment:
if I'm getting this right, perhaps the logic can be simplified? Just a minor comment


```
if(NOT IS_DIRECTORY "${CMAKE_OSX_SYSROOT}
   # this should cover the case for the Xcode generator,
   # assuming that the values here have a 1:1 with the `os.sdk` setting in Conan
   # we may want to restrict this to iOS/watchOS/tvOS,
   set(_OS_SDK  ${CMAKE_OSX_SYSROOT})
else()
    # see comment below
endif()
```

for the simulator ones, all of them have the `Simulator.platform` substring in the path, perhaps we can make it easier that way - although from what I can se, we need to map `iOS` -> `iphoneos`, `tvOS` -> `appletvos`, so maybe not straightforward

Reply:
I'll work on it a bit tomorrow, see if I can get something simpler than what I currently have.

Reply:
Your suggestion simplified the logic a bit, let me know what you think.